src/Parameters/ParameterTranslator.php

```diff
@@ -0,0 +1,140 @@
+<?php
+/**
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License
+ * as published by the Free Software Foundation; under version 2
+ * of the License (non-upgradable).
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ *
+ * Copyright (c) 2019 (original work) Open Assessment Technologies SA;
+ */
+
+namespace OAT\Library\DBALSpanner\Parameters;
+
+use Doctrine\DBAL\Exception\InvalidArgumentException;
+use Doctrine\DBAL\ParameterType;
+use Google\Cloud\Spanner\Database;
+use PhpMyAdmin\SqlParser\Lexer;
+use PhpMyAdmin\SqlParser\Token;
+
+class ParameterTranslator
+{
+    /** @var int */
+    private $positionalParameterCount = 0;
+
+    /**
+     * Detects the optional parameter syntax.
+     * DBAL and Spanner are both named so can be compatible.
+     * Named and positional syntaxes are not compatible.
+     *
+     * @param string $sql
```

Comment:
I don't think that we need this in PHP docs if we have type hinting for this param and you are not including any additional info for this param. 
```suggestion
```

Reply:
this will be as a clean-up , 
please gather all of the improvements ideas and put them inside the document listed inside the description of the PR